src/FullArticle.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { getArticles } from "./utils/getAllArticles";
+import { getArticleById } from "./utils/getArticleById";
+import { useParams } from "react-router-dom";
+import './ArticlePage.css'
+
+const FullArticle = () => {
+    const [article, setArticle] = useState(null)
+    const [isLoading, setIsLoading] = useState(true)
+    const { article_id } = useParams()
+
+    useEffect(() => {
+        getArticleById(article_id).then((data) => {
+            setArticle(data);
+            setIsLoading(false)
+        }).catch((err) => {
+            console.log(err)
+        })
+    }, [article_id])
```

Comment:
I believe we need to setLoading with true when we have anything in a dependancy array.This avoids loading being false when we are fetching the articles on the second invoation of useEffect.